integrationtests/tests/fixtures/real.rs

```diff
@@ -10,59 +8,104 @@ use async_trait::async_trait;
 use lightning_invoice::Invoice;
+use tokio::sync::Mutex;
 use tokio::time::sleep;
-use tonic_lnd::lnrpc::Invoice as LndInvoice;
-use tonic_lnd::{connect, LndClient};
+use tonic_lnd::lnrpc::{GetInfoRequest, Invoice as LndInvoice, ListChannelsRequest};
+use tonic_lnd::LndClient;
+use tracing::info;
 use url::Url;
 
+use super::GatewayNode;
 use crate::fixtures::LightningTest;
 
 #[derive(Clone)]
 pub struct RealLightningTest {
     rpc_cln: Arc<Mutex<ClnRpc>>,
     rpc_lnd: Arc<Mutex<LndClient>>,
     initial_balance: Amount,
+    // Which lightning node to use as the gateway?
+    gateway_node: GatewayNode,
     pub gateway_node_pub_key: secp256k1::PublicKey,
 }
 
 #[async_trait]
 impl LightningTest for RealLightningTest {
     async fn invoice(&self, amount: Amount, expiry_time: Option<u64>) -> Invoice {
-        let mut lnd_rpc = self.rpc_lnd.lock().await;
-        let tonic_invoice = match expiry_time {
-            Some(expiry) => LndInvoice {
-                value_msat: amount.msats as i64,
-                expiry: expiry as i64,
-                ..Default::default()
-            },
-            None => LndInvoice {
-                value_msat: amount.msats as i64,
-                ..Default::default()
-            },
-        };
-        let invoice_resp = lnd_rpc
-            .lightning()
-            .add_invoice(tonic_invoice)
-            .await
-            .unwrap()
-            .into_inner();
-
-        Invoice::from_str(&invoice_resp.payment_request).unwrap()
+        match self.gateway_node {
+            // If we're using CLN as the gateway, use LND to fetch an invoice
+            GatewayNode::Cln => {
+                info!("fetching invoice from lnd");
+                let mut lnd_rpc = self.rpc_lnd.lock().await;
+                let tonic_invoice = match expiry_time {
+                    Some(expiry) => LndInvoice {
+                        value_msat: amount.msats as i64,
+                        expiry: expiry as i64,
+                        ..Default::default()
+                    },
+                    None => LndInvoice {
+                        value_msat: amount.msats as i64,
+                        ..Default::default()
+                    },
+                };
+                let invoice_resp = lnd_rpc
+                    .lightning()
+                    .add_invoice(tonic_invoice)
+                    .await
+                    .unwrap()
+                    .into_inner();
+
+                Invoice::from_str(&invoice_resp.payment_request).unwrap()
+            }
+            // If we're using LND as the gateway, use CLN to fetch an invoice
+            GatewayNode::Lnd => {
+                info!("fetching invoice from cln");
+                let random: u64 = rand::random();
+                let invoice_req = requests::InvoiceRequest {
+                    amount_msat: AmountOrAny::Amount(ClnRpcAmount::from_msat(amount.msats)),
+                    description: "".to_string(),
```

Comment:
I haven't been following all the drama all twitter about this....but will the recent CLN description change break us if we use empty string here?

Reply:
let's just leave it here and update if CI ever breaks. this is just a test fixture so not a big deal.

Reply:
won't fix, for now